be/test/olap/schema_change_test.cpp

```diff
@@ -672,7 +672,505 @@ TEST_F(TestColumn, ConvertVarcharToDate) {
     OLAPStatus st = read_row.convert_from(1, read_row.cell_ptr(0), tp, _mem_pool.get());
     ASSERT_EQ(st, OLAP_ERR_INVALID_SCHEMA);
 }
-  
+
+TEST_F(TestColumn, ConvertVarcharToTinyInt) {
```

Comment:
Hi, Could we abstract the UT for ConvertVarcharToXXX ?

Reply:
> Hi, Could we abstract the UT for ConvertVarcharToXXX ?

I have tried use macro to abstrct in following, but It seems there is something wrong
```
#define TEST_CONVERT_FROM_VARCHAR(type_name, type_size, normal_value, overflow_value) \
TEST_F(TestColumn, ConvertFromVarchar) { \
    AddColumn( \
    "VarcharColumn", \
    "VARCHAR", \
    "REPLACE", \
    255, \
    false, \
    true); \

    AddColumn( \
    "ConvertColumn", \
    type_name, \
    "REPLACE", \
    type_size, \
    false, \
    false); \

    TabletSchema tablet_schema; \
    InitTablet(&tablet_schema); \
    CreateColumnWriter(tablet_schema); \

    RowCursor write_row; \
    write_row.init(tablet_schema); \

    RowBlock block(&tablet_schema); \
    RowBlockInfo block_info; \
    block_info.row_num = 10000; \
    block.init(block_info); \

    std::string src_str = normal_value; \
    write_row.set_field_content(0, src_str.data(), _mem_pool.get()); \
    block.set_row(0, write_row); \
    block.finalize(1); \
    ASSERT_EQ(_column_writer->write_batch(&block, &write_row), OLAP_SUCCESS); \

    ColumnDataHeaderMessage header; \
    ASSERT_EQ(_column_writer->finalize(&header), OLAP_SUCCESS); \

    CreateColumnReader(tablet_schema); \

    RowCursor read_row; \
    read_row.init(tablet_schema); \

    _col_vector.reset(new ColumnVector()); \
    ASSERT_EQ(_column_reader->next_vector(_col_vector.get(), 1, _mem_pool.get()), OLAP_SUCCESS); \
    char* data = reinterpret_cast<char*>(_col_vector->col_data()); \
    read_row.set_field_content(0, data, _mem_pool.get()); \
    const Field* src_field = read_row.column_schema(0); \
    read_row.convert_from(1, read_row.cell_ptr(0), src_field->type_info(), _mem_pool.get()); \
    std::string dst_str = read_row.column_schema(1)->to_string(read_row.cell_ptr(1)); \
    ASSERT_EQ(src_str, dst_str); \

    src_str = "invalid"; \
    write_row.set_field_content(0, src_str.data(), _mem_pool.get()); \
    block.set_row(0, write_row); \
    block.finalize(1); \
    ASSERT_EQ(_column_writer->write_batch(&block, &write_row), OLAP_SUCCESS); \

    _col_vector.reset(new ColumnVector()); \
    ASSERT_EQ(_column_reader->next_vector(_col_vector.get(), 1, _mem_pool.get()), OLAP_SUCCESS); \
    data = reinterpret_cast<char*>(_col_vector->col_data()); \
    read_row.set_field_content(0, data, _mem_pool.get()); \
    const Field* src_field2 = read_row.column_schema(0); \
    ASSERT_EQ(read_row.convert_from(1, read_row.cell_ptr(0), src_field2->type_info(), _mem_pool.get()), OLAP_ERR_INVALID_SCHEMA); \

    src_str = overflow; \
    write_row.set_field_content(0, src_str.data(), _mem_pool.get()); \
    block.set_row(0, write_row); \
    block.finalize(1); \
    ASSERT_EQ(_column_writer->write_batch(&block, &write_row), OLAP_SUCCESS); \

    _col_vector.reset(new ColumnVector()); \
    ASSERT_EQ(_column_reader->next_vector(_col_vector.get(), 1, _mem_pool.get()), OLAP_SUCCESS); \
    data = reinterpret_cast<char*>(_col_vector->col_data()); \
    read_row.set_field_content(0, data, _mem_pool.get()); \
    const Field* src_field3 = read_row.column_schema(0); \
    ASSERT_EQ(read_row.convert_from(1, read_row.cell_ptr(0), src_field3->type_info(), _mem_pool.get()), OLAP_ERR_INVALID_SCHEMA); \

    TypeInfo* tp = get_type_info(OLAP_FIELD_TYPE_HLL); \
    OLAPStatus st = read_row.convert_from(1, read_row.cell_ptr(0), tp, _mem_pool.get()); \
    ASSERT_EQ(st, OLAP_ERR_INVALID_SCHEMA); \
} \

TEST_CONVERT_FROM_VARCHAR("TINYINT", 1, "127", "128")
```